### PR TITLE
[Docker] Add `--gpus all` for docker when GPU is available

### PR DESCRIPTION
**Pull request by Michaelvll**
There seems no harm to add `--gpus all` when GPUs exist, which is a newer version of the option `--runtime nvidia` and seems docker's official doc is using this: https://docs.docker.com/engine/containers/resource_constraints/#access-an-nvidia-gpu 

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: sky/templates/aws-ray.yml.j2
Comment (cblmemo) [inline sky/templates/aws-ray.yml.j2:1]: should we remove other `--gpus all` in j2 files like `{azure,gcp,paperspace}-ray.yml.j2`?

**Review by cblmemo — COMMENTED** [review]
Will this affect the clouds that are still using `sky/skylet/providers/command_runner.py`? Those clouds are still using ray's docker configure code path iirc.

File: sky/templates/aws-ray.yml.j2
Comment (Michaelvll) [reply sky/templates/aws-ray.yml.j2:1]: Good catch! I just removed them. For those clouds still using node providers, they are not affected by the current PR and will still work as current master, which might be fine?

Comment (cblmemo) [reply sky/templates/aws-ray.yml.j2:1]: Oh actually, should we only remove `--gpus all` in clouds that support the new provisioner? For other clouds like paperspace, we should keep this in the ray file so the ray docker command runner can use this option to expose GPUs to the docker.

Comment (Michaelvll) [reply sky/templates/aws-ray.yml.j2:1]: paperspace is using new provisioner. : )

Comment (cblmemo) [reply sky/templates/aws-ray.yml.j2:1]: Oh good point! Then maybe we could remove the old docker command runner now.

**Review by cblmemo — APPROVED** [review]
Thanks for updating this @Michaelvll ! LGTM.

File: sky/templates/aws-ray.yml.j2
Comment (Michaelvll) [reply sky/templates/aws-ray.yml.j2:1]: Good point! Let's keep it for reference for now. We can probably remove it in a separate PR.

**Pull request merged**